test/lhash_test.c

```diff
@@ -573,7 +573,8 @@ static void do_mt_hash_work(void)
         }
         switch(behavior) {
         case DO_LOOKUP:
-            ossl_ht_read_lock(m_ht);
+            if (!ossl_ht_read_lock(m_ht))
+                break;
```

Comment:
Don't we want to fail here? i.e. set `worker_exits[num]` and return?

Reply:
we could, but we don't have to.  The lookups in this test just find an element in the hash table and confirm that its supposed to be there.  Failure to get a lock in this case just represents a transient failure that doesn't impact the outcome of what the test is actually testing.  It would be a problem if we could never get the lock for the duration of the test, that might be a problem, as it means we never tested a lookup operation.  But a single failure isn't really a problem.

Reply:
Why would this ever fail? Are there legitimate reasons why it might do?